lib/pages/dashboard/team_card.dart

```diff
@@ -0,0 +1,98 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:v34/commons/cards/titled_card.dart';
+import 'package:v34/commons/podium.dart';
+import 'package:v34/models/team.dart';
+import 'package:v34/pages/dashboard/blocs/team_classification_bloc.dart';
+import 'package:v34/repositories/repository.dart';
+
+class TeamCard extends StatefulWidget {
+  final Team team;
+  final bool active;
+  final double distance;
+  final double cardHeight = 190;
+
+  TeamCard({@required this.team, @required this.active, @required this.distance});
+
+  @override
+  _TeamCardState createState() => _TeamCardState();
+
+}
+
+class _TeamCardState extends State<TeamCard> {
+  TeamClassificationBloc _classificationBloc;
+
+  @override
+  void initState() {
+    super.initState();
+    _classificationBloc = TeamClassificationBloc(
+      repository: RepositoryProvider.of<Repository>(context)
+    );
+    if (widget.active) {
+      _classificationBloc.add(TeamClassificationLoadEvent(widget.team));
+    }
+  }
+
+  @override
+  void didUpdateWidget(TeamCard oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    // !oldWidget.active -> to avoid several calls to the API when sliding
+    // which can create a visual bug
+    if ((widget.team.clubCode != oldWidget.team.clubCode) || (widget.active && !oldWidget.active)) {
```

Comment:
Does it mean that when the club code change, you will fetch the classifications even if the widget is not active ? Is there a use case ?

BTW: `active` is not really meaningful, wdyt about `currentlyDisplayed` ? other ideas ?

Reply:
I can update the condition by adding the fact that the widget has to be active.

Reply:
Can you explain the issue you've seen to introduce this new condition ?

Reply:
`currentlyDisplayed` is a good idea ! I change it !

Reply:
> Can you explain the issue you've seen to introduce this new condition ?

This new condition permits the classification of the first team to be fetched when we choose another club with the first `PageView` whereas we still on the first `TeamCard` of the previous club. In this situation, the condition `oldWidget.active` is `true` and then the condition `widget.active && !oldWidget.active` is `false` so we don't fetch the data.

Reply:
Well spotted !